src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,83 @@
+import { createContext, useContext, useState, useEffect } from 'react';
+import type { ReactNode } from 'react';
+import type { User, AuthContextType } from '@/types/auth';
+import {
+  saveAuthToStorage,
+  loadAuthFromStorage,
+  removeAuthFromStorage,
+} from '@/utils/storage';
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+interface AuthProviderProps {
+  children: ReactNode;
+}
+
+export const AuthProvider = ({ children }: AuthProviderProps) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true); // 초기 로딩은 true로 시작
+
+  const isAuthenticated = !!user;
+
+  // 컴포넌트 마운트 시 localStorage에서 사용자 정보 복원
+  useEffect(() => {
+    const initializeAuth = async () => {
+      try {
+        const savedUser = loadAuthFromStorage();
+        if (savedUser) {
+          setUser(savedUser);
+        }
+      } catch (error) {
+        console.error('인증 정보 초기화 실패:', error);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    initializeAuth();
+  }, []);
```

Comment:
지금 user라는 상태를 선언하고 로컬스토리지에서 동기화하는 로직을 useEffect를 통해 구현하셨는데요,
로컬스토리지와 자동으로 동기화되는 리액트 state를 선언하는 커스텀훅을 만들면 로컬스토리지 동기화에 대한 로직을 숨길 수 있어 선언적으로 코드를 작성할 수 있을 것 같아요.

아래 형태로 user와 setUser를 사용할 수 있게끔 리팩터링해볼 수 있을까요?

```suggestion
  const [user, setUser] = useLocalStorageState<User | null>(
    '', // 로컬스토리지 key
    null // 로컬스토리지에 값이 없다면 초기값
  );
```

Reply:
개선해봤습니다 👍